File: backend/src/main/java/com/app/backend/dao/entities/Photo.java
package com.app.backend.dao.entities;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;


import java.util.Date;

@Document
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Builder
public class Photo {
    @Id
    private String  id;
    private byte[] file;
    private String  type;
    private boolean photoProfil;
    private Date shareDate;
    @DBRef
    private User user;

}

File: frontend/src/app/shared/Photo/photo.ts
import { User } from "../user";

export class Photo {
  id: string;
  name : string;
  file: any;
  type: string;
  photoProfil: boolean;
  shareDate: Date;
  user: User;

  constructor(
    id: string,
    name: string,
    file: any,
    type: string,
    photoProfil: boolean,
    shareDate: Date,
    user: User
  ) {
    this.id = id;
    this.name=name;
    this.file = file;
    this.type = type;
    this.photoProfil = photoProfil;
    this.shareDate = shareDate;
    this.user = user;
  }
}
